server/athenian/api/internal/features/metric_calculator.py

```diff
@@ -647,6 +647,58 @@ def reset(self) -> None:
         [c.reset() for c in self._calcs]
 
 
+class ThresholdComparisonRatioCalculator(AggregationMetricCalculator[float]):
+    """Calculate the ratio of metric values satisfying a given threshold.
+
+    Metric values are computed by the upstream MetricCalculator dependency.
+
+    Subclasses must specify the numpy comparison function
+    (eg. np.less_equal, np.greater, ...) to use between the metric
+    values and the threshold.
+
+    """
+
+    metric = MetricFloat
+
+    @abstractproperty
+    def compare(self) -> Callable:
+        """Compare a metric value against the threshold."""
```

Comment:
```suggestion
    @staticmethod
    def compare(peek: np.ndarray, threshold: Any, *, out: np.ndarray, where: np.ndarray) -> np.ndarray:
        """Compare a metric value against the threshold."""
        raise NotImplementedError
```
so that we specify the full usage contract of that callback.